src/history/hs_rec.c

```diff
@@ -495,11 +495,9 @@ __wt_hs_insert_updates(
          * away.
          */
         modify_cnt = 0;
-        for (; updates.size > 0 &&
-             !(upd->txnid == list->onpage_upd->txnid &&
-               upd->start_ts == list->onpage_upd->start_ts);
-             tmp = full_value, full_value = prev_full_value, prev_full_value = tmp,
-             upd = prev_upd) {
+        for (; updates.size > 0; tmp = full_value, full_value = prev_full_value,
+                                 prev_full_value = tmp, upd = prev_upd) {
+            WT_ASSERT(session, upd != list->onpage_upd);
```

Comment:
Can you explain this assert in a comment?

Reply:
Added a comment